alerting/src/test/kotlin/org/opensearch/alerting/transport/GetRemoteIndexesActionIT.kt

```diff
@@ -101,7 +101,9 @@ class GetRemoteIndexesActionIT : AlertingRestTestCase() {
 
             assertNotNull(clusterDetails[ClusterIndexes.INDEXES_FIELD])
             val indexes = clusterDetails[ClusterIndexes.INDEXES_FIELD] as Map<String, Map<String, Any>>
-            assertEquals(expectedNames.size, indexes.keys.size)
+
+            // Skipping this assert when security is enabled as it doesn't consider the existence of system indexes.
+            if (!securityEnabled()) assertEquals(expectedNames.size, indexes.keys.size)
```

Comment:
Nice find. We could try to compute the correct value when security is enabled rather than skipping this assert but that could end up being a moving target if new system indices are added down the line